rp2040-hal/src/watchdog.rs

```diff
@@ -122,17 +122,17 @@ impl watchdog::WatchdogEnable for Watchdog {
 
         // Due to a logic error, the watchdog decrements by 2 and
         // the load value must be compensated; see RP2040-E1
-        self.delay_ms = period.into().to_millis() * 2;
+        self.delay_us = period.into().to_micros() * 2;
```

Comment:
In theory this could overflow, which could be fixed by checking the max value before doing the multiplication.
In practice it probably doesn't matter because most firmwares will just hard code some sane value.